domain/credential/service/service_test.go

```diff
@@ -278,6 +278,27 @@ func (s *serviceSuite) TestCheckAndUpdateCredential(c *gc.C) {
 	}})
 }
 
+func (s *serviceSuite) TestCheckAndUpdateNewCredential(c *gc.C) {
```

Comment:
A comment here documenting that fact we are testing upsert logic and that if the credential does not exist this error is not bubbled up and the credential is instead created.